integration-tests/mock.ts

```diff
@@ -0,0 +1,51 @@
+import assert from "assert";
+import { blue, red } from "./base";
+import wcmatch from "wildcard-match";
+
+interface MockedFetch {
+    method: "GET" | "POST" | "PUT" | "PATCH";
+    url: string; // use some/path/* as a wildcard
+    body?: string; // Use * as a wildcard
+    responseStatus: number;
+    response?: string | object;
+}
+
+let expectedCalls: MockedFetch[] = [];
+
+export function expectFetch(mocked: MockedFetch) {
+    expectedCalls.push(mocked);
```

Comment:
Should we add a reset mechanism, so that every test starts with a clean state? Otherwise, test2 might still have the `expectFetch` elements from test1.

Reply:
good point, added a check